radsat-sk/operation/subsystems/RAntenna.c

```diff
@@ -0,0 +1,284 @@
+/**
+ * @file RAntenna.c
+ * @date Jan 15, 2022
+ * @author Addi Amaya (caa746)
+ */
+
+#include <RAntenna.h>
+#include <satellite-subsystems/IsisAntS.h>
+#include <string.h>
+#include <hal/errors.h>
+#include <hal/Timing/Time.h>
+
+/***************************************************************************************************
+                                  PRIVATE DEFINITIONS AND VARIABLES
+***************************************************************************************************/
+
+/** Struct for defining Antenna Systems I2C Addresses. Configured as a single bus, both address write to same location */
+ISISantsI2Caddress RAntennaI2Caddress = {ANTENNA_I2C_SLAVE_ADDR_PRIMARY,ANTENNA_I2C_SLAVE_ADDR_REDUNANT};
+
+/** Struct that holds the current deployment status of the antenna*/
+antenna_deployment_status_t RAntennaStatus = { 0 };
+
+/** Track whether the antenna has been initialized yet */
+static int antennaInitialized = 0;
+
+/** Track Number of times the antenna attempted to deploy*/
+static int antennaDeploymentAttempts = 0;
+
+
+/***************************************************************************************************
+                                             PUBLIC API
+***************************************************************************************************/
+
+/**
+ * Initializes the ISISpace Antenna driver
+ *
+ * @return 0 for success, non-zero for failure. See hal/errors.h for details.
+ */
+int antennaInit(void) {
+
+	// Only allow initialization once (return without error if already initialized)
+	if (antennaInitialized)
+		return 0;
+
+	// Isis Function call for antenna initialization
+	int error = IsisAntS_initialize(&RAntennaI2Caddress, ANTENNAS_ON_BOARD);
+
+	// Update flag if successfully initialized
+	if (error == 0)
+		antennaInitialized = 1;
+
+	// TODO: record errors (if present) to System Manager
+	return error;
+}
+
+/**
+ * Activate the auto-deployment mechanism on ISISpace Antenna System which will attempt to sequentially
+ * deploy all antennas present on the system without intervention. Has a built in timer per antenna deployment
+ * @return 0 for success, non-zero for failure. See hal/errors.h for details.
+ */
+int antennaDeploymentAttempt(void) {
+
+	// For Error detection
+	int error = 0;
+
+	// Initialize the union for antenna status
+	ISISantsStatus RISISantsStatus = { .fields = { 0 } };
+
+	// A Side deployment Attempt
+	while ( antennaDeploymentAttempts < MAX_DEPLOYMENT_ATTEMPTS ) {
+
+		// Get status of the A side antenna
+		error = IsisAntS_getStatusData(ANTENNA_I2C_SLAVE_ADDR_PRIMARY,isisants_sideA,&RISISantsStatus);
+
+		// Error check for requesting antenna status
+		if(error != 0) {
+
+			// TODO: record errors (if present) to System Manager
+			return error;
```

Comment:
One thought I just had, when an error is encountered like this, should the local state variables (e.g. RAntennaStatus) be reset at all? Is there any advantage or disadvantage to doing that? Just a thought.